lib/core/wallet_connect/web3wallet_service.dart

```diff
@@ -0,0 +1,208 @@
+
+    // Setup our accounts
+    List<ChainKeyModel> chainKeys = GetIt.I<WalletConnectKeyService>().getKeys();
+    for (final chainKey in chainKeys) {
+      for (final chainId in chainKey.chains) {
+        _web3Wallet!.registerAccount(
+          chainId: chainId,
+          accountAddress: chainKey.publicKey,
+        );
+      }
+    }
+
+    // Setup our listeners
+    log('web3wallet create');
+    _web3Wallet!.core.pairing.onPairingInvalid.subscribe(_onPairingInvalid);
+    _web3Wallet!.core.pairing.onPairingCreate.subscribe(_onPairingCreate);
+    _web3Wallet!.pairings.onSync.subscribe(_onPairingsSync);
+    _web3Wallet!.onSessionProposal.subscribe(_onSessionProposal);
+    _web3Wallet!.onSessionProposalError.subscribe(_onSessionProposalError);
+    _web3Wallet!.onSessionConnect.subscribe(_onSessionConnect);
+    _web3Wallet!.onAuthRequest.subscribe(_onAuthRequest);
+  }
+
+  @override
+  Future<void> init() async {
+    // Await the initialization of the web3wallet
+    log('web3wallet init');
+    await _web3Wallet!.init();
+
+    pairings.value = _web3Wallet!.pairings.getAll();
+    sessions.value = _web3Wallet!.sessions.getAll();
+    auth.value = _web3Wallet!.completeRequests.getAll();
+  }
+
+  @override
+  FutureOr<void> onDispose() {
+    log('web3wallet dispose');
+    _web3Wallet!.core.pairing.onPairingInvalid.unsubscribe(_onPairingInvalid);
+    _web3Wallet!.pairings.onSync.unsubscribe(_onPairingsSync);
+    _web3Wallet!.onSessionProposal.unsubscribe(_onSessionProposal);
+    _web3Wallet!.onSessionProposalError.unsubscribe(_onSessionProposalError);
+    _web3Wallet!.onSessionConnect.unsubscribe(_onSessionConnect);
+    _web3Wallet!.onAuthRequest.unsubscribe(_onAuthRequest);
+  }
+
+  @override
+  Web3Wallet getWeb3Wallet() {
+    return _web3Wallet!;
+  }
+
+  void _onPairingsSync(StoreSyncEvent? args) {
+    if (args != null) {
+      pairings.value = _web3Wallet!.pairings.getAll();
+    }
+  }
+
+  void _onSessionProposalError(SessionProposalErrorEvent? args) {
+    log(args.toString());
+  }
+
+  void _onSessionProposal(SessionProposalEvent? args) async {
+    if (args != null) {
+      final Widget modalWidget = Web3RequestModal(
+        child: ConnectionRequestWidget(
+          wallet: _web3Wallet!,
+          sessionProposal: SessionRequestModel(request: args.params),
+        ),
+      );
+      // show the bottom sheet
+      final bool? isApproved = await _bottomSheetHandler.queueBottomSheet(
+        widget: modalWidget,
+      ) as bool?;
+
+      if (isApproved != null && isApproved) {
+        _web3Wallet!.approveSession(
+          id: args.id,
+          namespaces: args.params.generatedNamespaces!,
+        );
+      } else {
+        _web3Wallet!.rejectSession(
+          id: args.id,
+          reason: Errors.getSdkError(
+            Errors.USER_REJECTED,
+          ),
+        );
+      }
+    }
+  }
+
+  void _onPairingInvalid(PairingInvalidEvent? args) {
+    log('Pairing Invalid Event: $args');
```

Comment:
shouldn't we show a message for the user here?

Reply:
Yes, added a dialog popup

Reply:
Yes we should, added a bottom sheet display for the message.